Server/BackEndMonitoringServer/src/CServiceHandler.cpp

```diff
@@ -0,0 +1,287 @@
+                service_control_manager->GetHandle(), 
+                m_service->GetName(),
+                SERVICE_QUERY_STATUS | 
+                SERVICE_START));
+
+        if (service->GetHandle() == nullptr)
+        {
+            Utils::DisplayError("Failed to open the service");
+            success = false;
+            break;
+        }
+
+        if (StartService(service->GetHandle(), 0, NULL) == 0)
+        {
+            Utils::DisplayError("Failed to start the service");
+            success = false;
+            break;
+        }
+    } while (false);
+    
+    success ? Utils::DisplayMessage("Service started")
+            : Utils::DisplayMessage("Failed to start the service");
+
+    return success;
+}
+
+bool ServiceHandler::Stop() const
+{
+    Utils::DisplayMessage("Stopping the service");
+
+    bool success = true;
+
+    do
+    {
+        auto service_control_manager = std::make_unique<ServiceHandle>(
+            ::OpenSCManager(
+                nullptr, nullptr,
+                SC_MANAGER_CONNECT));
+
+        if (service_control_manager->GetHandle() == nullptr)
+        {
+            Utils::DisplayError("Failed to open the service control manager");
+            success = false;
+            break;
+        }
+
+        auto service = std::make_unique<ServiceHandle>(
+            ::OpenService(
+                service_control_manager->GetHandle(), 
+                m_service->GetName(),
+                SERVICE_QUERY_STATUS | 
+                SERVICE_STOP));
+
+        if (service->GetHandle() == nullptr)
+        {
+            Utils::DisplayError("Failed to open the service");
+            success = false;
+            break;
+        }
+
+        SERVICE_STATUS status = {};
+        if (::ControlService(service->GetHandle(), SERVICE_CONTROL_STOP, &status) != 0)
+        {    
+            while (::QueryServiceStatus(service->GetHandle(), &status) != 0)
+            {
+                if (status.dwCurrentState != SERVICE_STOP_PENDING)
+                {
+                    // TODO: Add information
+                    break;
+                }
+            }
+
+            if (status.dwCurrentState != SERVICE_STOPPED)
+            {
+                success = false;
+                break;
+            } 
+        } 
+        else
+        {
+            Utils::DisplayError("Didn't control service");
+        } 
+    } while (false);
+    
+    success ? Utils::DisplayMessage("Service stopped")
+            : Utils::DisplayMessage("Failed to stop the service");
+
+    return success;
+}
+
+bool ServiceHandler::Restart() const
+{
+    Utils::DisplayMessage("Starting the service");
+
+    bool success = true;
+
+    if (success = Stop())
+    {
+        success = Start();
+    }
+    
+    success ? Utils::DisplayMessage("Service restarted")
+            : Utils::DisplayMessage("Failed to restart the service");
+
+    return success;
+}
+
+bool ServiceHandler::Run() const
+{
+    return m_service->Run();
```

Comment:
I couldn't find header, but, as I see, m_service - pointer. Then rename it into m_p_service

Reply:
Fixed